### PR TITLE
It/scored

### LINKED ISSUE #1993: Configure verification-enabled streams
- Read `VerificationFreq` from auth-webhook response for a particular stream
- Read trust score for each orchestrator from orch-webhook response
- Use the capability signalling mechanism between B <> O to make sure we send verification enabled streams to Os that support generating mpeg7 signatures

Refer to steps 1, 2 and 8 in the spec #1991 for details

### LINKED ISSUE #1994: Use multiple orchestrators for verification-enabled streams
- Select multiple orchestrators for verification-enabled streams
- Send the segment for transcoding w/ signature generation to one or many Os

Refer to the steps 3 and 4 in the spec #1991 for details

### PR DESCRIPTION
**Pull request by darkdarkdragon**
**What does this pull request do? Explain your changes. (required)**
Adds 'score' to orchestrator, separate trusted/untrusted orchestrators by score, allows to send same segment to different orchestrators simultaneously and compare hashes of results.

**Specific updates (required)**
- Read `VerificationFreq` from auth web hook response, save it to `StreamParameters`
- Read 'Score' field from orch web hook response, save it per-orchestrator
- Separate session pool management from `BroadcastSessionsManager` to separate struct `SessionPool`
- `BroadcastSessionsManager` now contains separate pools for trusted orchestrators and for untrusted ones
- If `VerificationFreq` field is zero for stream, transcoding path doesn't change - `BroadcastSessionsManager` finds one session (first tries trusted pool, if not found in trusted - check untrusted one) and uses it for transcoding as usual
- If `VerificationFreq` is greeted than zero - then `BroadcastSessionsManager` tries to find session in trusted pool and two sessions in untrusted, sends segments to all found session, then downloads hashes from all orchestrators, and if hash from untrusted orchestrator matches one from trusted - downloads transcoded segments from untrusted. 

**How did you test each of these updates (required)**
- Added unit tests
- Manual local testing

**Does this pull request close any open issues?**
- Fixes #1993 
- Fixes #1994 

**Checklist:**

- [ ] Read the [contribution guide](./doc/contributing.md)
- [ ] `make` runs successfully
- [ ] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: core/capabilities_test.go [hunk 1]
```diff
@@ -147,6 +147,16 @@ func TestCapability_JobCapabilities(t *testing.T) {
...
 
+	// check MPEG7VideoSignature
+	params.VerificationFreq = 1
+	assert.True(checkSuccess(params, []Capability{
+		Capability_H264,
+		Capability_MPEGTS,
+		Capability_FractionalFramerates,
+		Capability_AuthToken,
+		Capability_MPEG7VideoSignature,
+	}), "failed with fractional framerates")
```
Comment (jailuthra) [inline core/capabilities_test.go]:
```suggestion
	}), "failed with fast verification enabled")
```

File: common/types.go [hunk 2]
```diff
@@ -18,10 +48,47 @@ type CapabilityComparator interface {
...
-	GetOrchestrators(int, Suspender, CapabilityComparator) ([]*net.OrchestratorInfo, error)
+	// GetInfo gets info for specific orchestrator
+	GetInfo(uri string) OrchestratorLocalInfo
+	GetInfos() []OrchestratorLocalInfo
+	GetOrchestrators(int, Suspender, CapabilityComparator, ScorePred) ([]*net.OrchestratorInfo, error)
 	Size() int
+	SizeWithPred(ScorePred) int
+}
+
+func SizePredAtLeast(minScore float32) ScorePred {
```
Comment (jailuthra) [inline common/types.go]: Nitpick:

The current naming convention makes code hard to read
```go
node.OrchestratorPool.SizeWithPred(common.SizePredAtLeast(common.Score_Trusted))
```

Given the predicate function returned here doesn't care about the pool size, I find `ScoreAtLeast` a much easier to read name.

Similarly we can change `SizePredEqual -> ScoreEqualTo`, and also change size method of interface `SizeWithPred -> SizeWith`.

So the calls would read like:
```go
node.OrchestratorPool.SizeWith(common.ScoreAtLeast(common.Score_Trusted))
selectOrchestrator(node, params, trustedNumOrchs, susTrusted, common.ScoreEqualTo(common.Score_Untrusted))
```

File: server/broadcast.go [hunk 3]
```diff
@@ -625,61 +911,7 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 			errCode = subType
 		}
 	}
-
-	// [EXPERIMENTAL] send content detection results to callback webhook
-	if DetectionWebhookURL != nil && len(res.Detections) > 0 {
-		glog.V(common.DEBUG).Infof("Got detection result %v", res.Detections)
-		go func(mid core.ManifestID, config core.DetectionConfig, seqNo uint64, detections []*net.DetectData) {
```
Comment (jailuthra) [inline server/broadcast.go:632]: Did this get moved to a subroutine somewhere? (if yes, i can't seem to find it)

File: server/segment_rpc.go [hunk 4]
```diff
@@ -457,6 +457,7 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64)
 	}
 
 	glog.Infof("Submitting segment nonce=%d manifestID=%s sessionID=%s seqNo=%d bytes=%v orch=%s timeout=%s", nonce, params.ManifestID, sess.OrchestratorInfo.AuthToken.SessionId, seg.SeqNo, len(data), ti.Transcoder, dur)
+	glog.Errorf("Submitting segment nonce=%d manifestID=%s sessionID=%s seqNo=%d bytes=%v orch=%s timeout=%s", nonce, params.ManifestID, sess.OrchestratorInfo.AuthToken.SessionId, seg.SeqNo, len(data), ti.Transcoder, dur)
```
Comment (jailuthra) [inline server/segment_rpc.go]: Can remove this if not necessary outside debugging

File: server/broadcast.go [hunk 5]
```diff
@@ -73,61 +70,180 @@ func (cfg *BroadcastConfig) SetMaxPrice(price *big.Rat) {
...
+	}
+	return res
+}
+
+func selectSession(sessions []*BroadcastSession, exclude []*BroadcastSession, durMult int) *BroadcastSession {
+	for _, session := range sessions {
+		if len(session.SegsInFlight) > 0 &&
+			time.Since(session.SegsInFlight[0].startTime) < time.Duration(durMult)*session.SegsInFlight[0].segDur &&
+			!includesSession(exclude, session) {
+			// Re-use last session if oldest segment is in-flight for < segDur
```
Comment (jailuthra) [inline server/broadcast.go]:
```suggestion
			// Re-use last session if oldest segment is in-flight for < durMult * segDur
```

**Review by jailuthra — COMMENTED** [review]
New design is much cleaner, and mostly LGTM!

File: server/segment_rpc.go (hunk 4 above)
Comment (darkdarkdragon) [reply server/segment_rpc.go]: Fixed

File: server/broadcast.go (hunk 3 above)
Comment (darkdarkdragon) [reply server/broadcast.go:632]: Lost somewhere when was merging 😄 
Put it back d943b851040a159354b857cd92601dca396aaa27
Added unit test.

File: common/types.go (hunk 2 above)
Comment (darkdarkdragon) [reply common/types.go]: Fixed

File: core/capabilities_test.go (hunk 1 above)
Comment (darkdarkdragon) [reply core/capabilities_test.go]: Fixed

**Pull request merged**